### PR TITLE
Implement `SubTask.getOwnerExecutable`

### PR DESCRIPTION
**Pull request by jglick**
Implements the API introduced in https://github.com/jenkinsci/jenkins/pull/7599, for now without requiring a newer core dep.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java [hunk 1]
```diff
@@ -429,6 +429,7 @@ public String getCookie() {
             return j.getNode(label);
         }
 
+        @Deprecated
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java:432]: Not really related, just noticed while I was here.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java [hunk 2]
```diff
@@ -721,7 +736,7 @@ private String computeEnclosingLabel(FlowNode executorStepNode, List<FlowNode> h
         }
 
         @Override public long getEstimatedDuration() {
-            Run<?,?> r = run();
+            Run<?,?> r = runForDisplay();
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java:739]: Also only tangentially related—there was no apparent reason not to be calling the newer and more robust `runForDisplay` in this context.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java [hunk 3]
```diff
@@ -613,6 +613,7 @@ private static void assertLogMatches(WorkflowRun build, String regexp) throws IO
                 Queue.Item[] items = Queue.getInstance().getItems();
                 assertEquals(1, items.length);
                 assertEquals(p, items[0].task.getOwnerTask());
+                assertEquals(b, items[0].task.getOwnerExecutable());
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java:616]: Test is covering something unrelated, but seemed as good a place as any to cover the new method.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java [hunk 4]
```diff
@@ -535,6 +536,10 @@ public String getShortDescription() {
             return hasPermission(Item.CANCEL);
         }
 
+        /**
+         * @deprecated use {@link #getOwnerExecutable} (which does not require a dependency on this plugin).
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java:540]: Potentially confusing without a core bump, as a dependency on this plugin or core ~2.389 is needed.

Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java:540]: True. Will keep this in draft unless & until the core change is merged and I can add a version number.

**Review by car-roll — APPROVED** [review]

**Pull request merged**